.github/workflows/ci-pyright-fails.yml

```diff
@@ -34,4 +34,4 @@ jobs:
       - name: Pyright
         uses: jakebailey/pyright-action@v1
         with:
-          version: 1.1.299
+          version: 1.1.300
```

Comment:
I propose to keep the version fixed so we have control over updating the version

Reply:
Could we actually integrate this with something like dependabot thing? Something that opens a PR on a new version, but does not directly merge it. Maybe if we stopped using the pyright action, made pyright an optional requirement, and then call it from the cmd line.

Reply:
Sounds like another issue to me :)

Reply:
#630